xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/doc/XWikiDocument.java

```diff
@@ -5441,23 +5421,7 @@ public void addAttachment(XWikiAttachment attachment)
     public XWikiAttachment setAttachment(XWikiAttachment attachment)
     {
         attachment.setDoc(this);
```

Comment:
Should probably move this inside XWikiAttachmentList#set() to be sure all added attachments have the right document.

Reply:
Actually you do have this feature in XWikiAttachmentList arelady but you do it after adding the attachment instead of before :)